src/mono/mono/mini/monovm.c

```diff
@@ -275,41 +270,45 @@ monovm_runtimeconfig_initialize (MonovmRuntimeConfigArguments *arg, MonovmRuntim
 }
 
 void
-mono_extract_and_register_properties (const char *buffer, long fileSize)
+mono_extract_and_register_properties (const char *buffer, guint64 file_size)
 {
-	int propertyCount;
-	int strLen;
-	int currentIdx = 0;
-	char *propertyKeys [MAX_PROPERTY_COUNT];
-	char *propertyValues [MAX_PROPERTY_COUNT];
-
-	propertyCount = buffer [0] - '\0';
-	currentIdx = 1;
-	for (int i = 0; i < propertyCount; ++i)
+	int property_count;
+	int str_len;
+	int current_idx = 0;
+	char **property_keys;
+	char **property_values;
+
+	property_count = mono_metadata_decode_value(buffer, &buffer);
+	property_keys = g_new0 (char*, property_count);
+	property_values = g_new0 (char*, property_count);
+	current_idx++;
```

Comment:
I don't think you need `current_idx`. It is just `buffer - buffer_start` (and you can compute it when you need it in the `g_asserts` below.

Also `mono_metadata_decode_value` technically moves the pointer by between 1 and 4 bytes - it depends on the value.  if it's small it's stored in 1 byte, if it's bigger it's 2 or 4.  so `current_idx` is calculated incorrectly here.

Reply:
`current_idx` is used to check if there is enough data in `buffer` before dereference it.

Reply:
Maybe I was over thinking...